### DISCUSSION TITLE
Keine Homematic Geräte und Funkstick mehr nach OS Update

### DISCUSSION THREAD

**Discussion by skyflash13** [SukramJ/aiohomematic#1444] [Q&A] [answered]
Liebe Community und Programmierer,

seit mehren Tagen funktioniert mein Homematic in Homeassistant nicht mehr.

Ich nutze RaspberryMatic mit der Homematic IP Local Integration. Bis vor kurzen hat noch alles normal funktioniert.
Nach einem HAOS Update auf 11.5 kam plötzlich diese Fehlermeldung:

```
Logger: hahomematic.central
Source: runner.py:188
First occurred: 17. Februar 2024 um 16:14:37 (48754 occurrences)
Last logged: 21:50:51

CHECK_CONNECTION failed: No clients exist. Trying to create clients for server RaspberryMatic
CREATE_CLIENTS failed: No connection to interface RaspberryMatic-BidCos-RF [OSError on RaspberryMatic-BidCos-RF: (111, 'Connection refused')
```

Alle Geräte funktionieren im Raspberrymatic Addon normal nur nach HA werden sie nicht durchgereicht.

Nach dem Fehler habe ich den HA Core upgedatet und auch die Homematic Integration. Leider ohne Erfolg.
Da HA beim OS Update kein Backup anlegt kann ich auch nicht zurück.

Ich nutze einen normalen Intel PC mit:
HAOS 11.5
Core 2024.2.2
Homematic IP Local 1.56.0
RaspberyMatic 3.73.9.20240130 

Der Stick ist ein HM-CFG-USB-2

Hat jemand auch das Problem oder noch besser eine Lösung?

Ich hoffe es kann mir hier wer helfen.

Danke im Voraus und LG
Andy

**Comment by SukramJ** [comment]
Die Integration hat keine Verbindung zum Addon. Also Addon und ggf. den host mal durchstarten.

**Comment by skyflash13** [comment]
Hallo SukramJ,

habe das ganze System schon mehrmals neu gestartet, hat leider nichts geändert.

Mir ist nur aufgefallen das sich anscheinend die IP Adresse von Raspberrymatic verändert hat. Wenn ich die in Homematic IP Eintrage und absende sagt er auch wieder keine Verbindung und setzt auf die alte Adresse zurück.

**Reply by SukramJ** [reply]
> Mir ist nur aufgefallen das sich anscheinend die IP Adresse von Raspberrymatic verändert hat

Des wegen steht ja auch in den Anleitungen, das man den Hostnamen und nicht die IP-Adresse verwenden soll.

**Reply by skyflash13** [reply]
Hallo SukramJ,

das habe ich auch gemacht. Aber jetzt steht nicht mehr der Name sondern die IP dort.

**Reply by SukramJ** [reply]
Welche Version hast du installiert?

**Reply by SukramJ** [reply, marked as answer]
Welche Verzeichnisse hast du unter custom_components?
hahm und homematicip_local?

**Reply by skyflash13** [reply]
Habe nur homematicip_local - Versionsnummer 1.56.0
Zumindest steht das bei Benutzerdefinierte Repositories - sonst sehe ich die Versionsnummer nirgends.
![homematicip_2](https://github.com/danielperna84/hahomematic/assets/122051742/5575ae87-beb4-4ab5-bc9d-0eebe879a2aa)
![homematicip](https://github.com/danielperna84/hahomematic/assets/122051742/1594a0e8-5cff-4d48-b852-391a3e9a4469)

**Reply by SukramJ** [reply]
Die Frage bezog sich eigentlich auf das Dateisystem.
Hänge bitte die Diagnoseinformationen an. Auf die Integration klicken und Diagnose Informationen herunterladen.
Gibt es Fehler in den AddOn Logs?

**Reply by skyflash13** [reply]
Ich habe jetzt mehrmals versucht den Host oder auch die IP Adresse einzugeben. Den Fehler hänge ich hier an.
In welcher Datei stehen die Einträge, vielleicht kann ich sie ja dort ändern.
![fehler](https://github.com/danielperna84/hahomematic/assets/122051742/7805367f-0a8e-48b8-8a51-87ac50b136f5)
![raspberrymatic](https://github.com/danielperna84/hahomematic/assets/122051742/3f9a9ae3-31f5-4739-bb74-bdf9ae572125)

**Reply by skyflash13** [reply]
Anbei die Diagnose Information
[config_entry-homematicip_local-694caed4126cd1172fa47f7a4d08a961 (1).json](https://github.com/danielperna84/hahomematic/files/14390670/config_entry-homematicip_local-694caed4126cd1172fa47f7a4d08a961.1.json)

**Comment by skyflash13** [comment]
Bei den Addon Logs gibt es keines von Homematic nur Raspberrymatic und das zeigt keinen Fehler - siehe Anhang.
[de838cd8_raspberrymatic_2024-02-23T22-44-15.297Z.log](https://github.com/danielperna84/hahomematic/files/14390685/de838cd8_raspberrymatic_2024-02-23T22-44-15.297Z.log)

**Comment by skyflash13** [comment]
Das Einzige das mir noch auffällt ist das in der Einstellung von Homematic die IP 172.30.33.1 steht obwohl es laut Raspberrymatic 172.30.33.2 ist. Ich habe dort aber ursprünglich den Host Namen eingegeben. Auch wenn ich das jetzt ändere funktioniert es nicht und am Ende steht immer die Falsche IP drinnen.
![ipadresse](https://github.com/danielperna84/hahomematic/assets/122051742/96e9d48f-c81a-41f0-9535-17b058334da2)

**Reply by skyflash13** [reply]
Hier noch das Ergebnis von Host in der Console:
 
![host](https://github.com/danielperna84/hahomematic/assets/122051742/9fcaa912-c8c8-4fbd-af7a-e488cc81cb52)

**Reply by SukramJ** [reply]
In der Fehlermeldung steht, das Du einen Hostnamen mit Leerzeichen eingetragen hast.
Im Screenshot ist ein führendes Leerzeichen zu sehen.

**Reply by SukramJ** [reply]
Bitte das HA log(Full log) anhängen.
Einstellungen- System- Protokolle (vollständige Protokole laden)

**Reply by skyflash13** [reply]
Habe nochmal probiert und kann kein Leerzeichen "sehen". Anbei das Log.
[Uploading home-assistant_2024-02-24T15-50-57.437Z.log…]()

**Reply by SukramJ** [reply]
Der Link funktioniert nicht. Der verweist nur wieder auf dieses Issue.
Das führende Leerzeichen ist sehr gut im Screenshot von heute morgen zu sehen.

**Comment by skyflash13** [comment]
Das Leerzeichen sehe ich im Screenshot. 

Wenn ich in Homematic IP auf Konfigurieren gehe und dort den Host nochmals Eintrage - (ohne Leerzeichen)
![konfig1](https://github.com/danielperna84/hahomematic/assets/122051742/e1c60e25-92ba-4e8f-9b54-d5a53f2add22)

![konfig2](https://github.com/danielperna84/hahomematic/assets/122051742/aa8b8812-caf7-447d-ab80-ed1b694cca67)

Und dann Absende kommt die Fehlermeldung
![konfig3](https://github.com/danielperna84/hahomematic/assets/122051742/96d6764f-dfd9-463a-acd8-1b803d25c514)

Und wenn ich es dann wieder öffne steht wieder die falsche IP dort
![konfig4](https://github.com/danielperna84/hahomematic/assets/122051742/ae1d8d17-7957-4180-8d55-1569cc16d263)

In welcher Datei steht der Eintrag? Kann ich ihm dort ändern?

Hier nochmal der Versuch mit dem Log.
[home-assistant_2024-02-25T08-35-14.707Z.log](https://github.com/danielperna84/hahomematic/files/14395751/home-assistant_2024-02-25T08-35-14.707Z.log)

Vielen Dank für Deine Unterstützung.

**Reply by SukramJ** [reply]
Kann der HM-CFG-USB-2 überhaupt HomematicIP? Ich glaube nicht, also solltest Du HomematicIP auch nicht in der Konfiguration auswählen, da der entsprechende Server auf der CCU nicht gestartet wird.
Der Eintrag wird gespeichert, wenn die Verbindung erfolgreich aufgebaut werden konnte.

Vorm nächsten Versuch mal bitte HA neustarten, dann wird auch das Log geleert, und ist übersichtlicher.

**Comment by skyflash13** [comment]
Lieber SukramJ,

ich habe jetzt HA neu gestartet und die Konfig nochmal probiert, diesmal ohne Homematic IP Hackerl und es funktioniert auf einmal wieder.

Hurra - Ich verstehe zwar nicht warum - aber es geht jetzt wieder.

Viel lieben Dank für Deine Zeit und Hilfe.

**Reply by SukramJ** [reply]
Sehr schön. Die Erklärung hatte ich ja [hier](https://github.com/danielperna84/hahomematic/discussions/1444#discussioncomment-8581809) geschrieben.

**Reply by skyflash13** [reply]
Versteh- super und nochmals vielen Dank für Deine Unterstützung.
